src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        joinedStr += `${this[i]}`;
+      }
+
+      if (i + 1 === this.length) {
+        break;
+      }
+
+      joinedStr += `${separator}`;
```

Comment:
```suggestion
      if (i < this.length - 1) {
        joinedStr += separator;
      }
```

Reply:
I followed the rule "not write the main code in IF "

Reply:
Here we have only 1 line of code in `if` block, so it's ok
In my opinion, my suggestion is easier for reading and understanding. If you want you can leave it as it is, it's not critical, it's just a recommendation to make your code a little bit easier for understanding

Reply:
Ok, thank you!